pokemon-showdown

```diff
@@ -22,6 +24,84 @@ try {
 	child_process.execSync('npm install --production', {stdio: 'inherit'});
 }
 
+// Check if the server is configured to use Go, and ensure the required
+// environment variables and dependencies are available if that is the case
+
+let config;
+try {
+	config = require('./config/config');
+} catch (e) {}
+
+if (config && config.golang) {
+	if (!process.env.GOPATH) {
+		console.log('The GOPATH environment variable is not set! It is required in order to run the server using Go.');
+		process.exit(0);
+	}
+	if (!process.env.GOROOT) {
+		console.log('The GOROOT environment variable is not set! It is required in order to run the server using Go.');
+		process.exit(0);
+	}
+
+	const dependencies = ['github.com/gorilla/mux', 'github.com/igm/sockjs-go/sockjs'];
+	let packages = child_process.execSync('go list all', {stdio: null, encoding: 'utf8'});
+	for (let dep of dependencies) {
+		if (!packages.includes(dep)) {
+			console.log(`Dependency ${dep} is not installed. Fetching...`);
+			child_process.execSync(`go get ${dep}`, {stdio: 'inherit'});
+		}
+	}
+
+	const {GOPATH} = process.env;
+	let stat;
+	let needsSrcDir = false;
+	try {
+		stat = fs.lstatSync(path.resolve(GOPATH, 'src/github.com/Zarel'));
+	} catch (e) {
+		needsSrcDir = true;
+	} finally {
+		if (stat && !stat.isDirectory()) {
+			needsSrcDir = true;
+		}
+	}
+
+	let srcPath = path.resolve(__dirname, 'sockets');
+	let tarPath = path.resolve(GOPATH, 'src/github.com/Zarel/Pokemon-Showdown/sockets');
+	if (needsSrcDir) {
+		try {
+			fs.mkdirSync(path.resolve(GOPATH, 'src/github.com/Zarel'));
+			fs.mkdirSync(path.resolve(GOPATH, 'src/github.com/Zarel/Pokemon-Showdown'));
+		} catch (e) {
+			console.log(e);
+			console.error(`Cannot make go source directory for the sockets library files! Symlink them manually from ${srcPath} to ${tarPath}`);
+			process.exit(0);
+		}
+	}
+
+	try {
+		stat = fs.lstatSync(path.resolve(GOPATH, 'src/github.com/Zarel/Pokemon-Showdown/sockets'));
+	} catch (e) {}
+
+	if (!stat || !stat.isSymbolicLink()) {
+		try {
+			if (process.platform === 'win32') {
+				child_process.execSync(`mklink /J ${tarPath} ${srcPath}`, {stdio: 'inherit'});
```

Comment:
First time I tried that for Windows symlinks it required admin access, but I didn't realize junctions didn't at the time, which is dumb since I'm already using junctions here. I'll need to work out a way to detect whether the junction exists though, since it's not recognized as a symbolic link by `stat`